### PR TITLE
fixes #90

### LINKED ISSUE #90: Error message for source-expression containing invalid characters should be verbous.
Ex. https://cspvalidator.org/#headerValue[]=script-src+%D0%B0&strategy=intersection . 
Say that chars outside ASCII 0x21-0x7E must be percent encoding or something like that

### PR DESCRIPTION
**Pull request by shekyan**
verbose error in tokeniser for bad characters in directive-value

### CODE REVIEW

File: src/main/java/com/shapesecurity/salvation/Tokeniser.java [hunk 1]
```diff
@@ -14,7 +16,7 @@
     private static final Pattern directiveSeparator = Pattern.compile(";");
     private static final Pattern policySeparator = Pattern.compile(",");
     private static final Pattern directiveNamePattern = Pattern.compile("[a-zA-Z0-9-]+");
-    private static final Pattern directiveValuePattern = Pattern.compile("[^;,\0- \\x7F]+");
+    private static final Pattern directiveValuePattern = Pattern.compile("[!-+--:<-~]+");
```
Comment (michaelficarra) [inline src/main/java/com/shapesecurity/salvation/Tokeniser.java]: This RegExp is the greatest.

File: src/main/java/com/shapesecurity/salvation/Tokeniser.java [hunk 2]
```diff
@@ -5,7 +5,9 @@
 
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
+import java.nio.charset.StandardCharsets;
 import java.util.ArrayList;
+import java.util.Arrays;
```
Comment (michaelficarra) [inline src/main/java/com/shapesecurity/salvation/Tokeniser.java]: Are these new imports used? They don't appear to be.

File: src/main/java/com/shapesecurity/salvation/Tokeniser.java [hunk 3]
```diff
@@ -98,7 +100,8 @@ private String next() {
                 continue;
             while (this.hasNext()) {
                 if (!this.eatDirectiveValue()) {
-                    throw this.createError("expecting directive-value but found " + this.next());
+                    int cp = this.next().codePointAt(0);
```
Comment (michaelficarra) [inline src/main/java/com/shapesecurity/salvation/Tokeniser.java]: Even though it's safe to do so, I'd prefer we called `this.next()` only once.

File: src/main/java/com/shapesecurity/salvation/Tokeniser.java [hunk 4]
```diff
@@ -98,7 +100,8 @@ private String next() {
...
+                    throw this.createError(String.format("expecting directive-value but found %s (U+%04x). Non-whitespace characters outside ASCII 0x21-0x7E must be percent-encoded", this.next(), cp));
```
Comment (michaelficarra) [inline src/main/java/com/shapesecurity/salvation/Tokeniser.java]: Capitalise the message. Might also want to just say non-ASCII and non-printable characters. That's simpler than specifying the code point range.

**Comment by shekyan** [comment]
Check it out.

**Pull request merged**